File: Infrastructure/style/examples/r/generate_examples.R
#!/usr/bin/env Rscript

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)
if (length(file_arg) == 0) {
  stop("Cannot resolve script path.", call. = FALSE)
}

script_path <- sub("^--file=", "", file_arg[1])
script_path <- normalizePath(script_path, mustWork = FALSE)
repo_root <- normalizePath(file.path(dirname(script_path), "..", ".."), mustWork = FALSE)

source(file.path(repo_root, "themes", "r", "eig_theme.R"))
tokens <- eig_load_tokens(file.path(repo_root, "themes", "r", "eig_tokens.R"))
eig_assert_fonts(tokens, allow_fallback = TRUE)

out_dir <- file.path(repo_root, "examples", "outputs", "r")
dir.create(out_dir, recursive = TRUE, showWarnings = FALSE)

set.seed(42)
n <- 140
weight <- pmax(pmin(rnorm(n, mean = 3.1, sd = 0.65), 5.7), 1.5)
cyl <- sample(c(4, 6, 8), n, replace = TRUE, prob = c(0.4, 0.35, 0.25))
origin <- sample(c("Domestic", "Foreign"), n, replace = TRUE, prob = c(0.58, 0.42))
mpg <- 41.5 - 5.0 * weight - 0.75 * (cyl - 4) + rnorm(n, sd = 2.1)
df <- data.frame(weight = weight, mpg = mpg, cyl = factor(cyl), origin = factor(origin))

old_par <- eig_base_defaults(tokens)
png(file.path(out_dir, "base_scatter.png"), width = 1400, height = 900, res = 180)
plot(
  df$weight,
  df$mpg,
  pch = 16,
  col = tokens$EIG_COLORS[["eig_teal_900"]],
  xlab = "Weight",
  ylab = "MPG",
  main = "Fuel Economy by Vehicle Weight"
)
dev.off()
par(old_par)

if (requireNamespace("ggplot2", quietly = TRUE)) {
  p <- ggplot2::ggplot(df, ggplot2::aes(weight, mpg, color = cyl)) +
    ggplot2::geom_point(size = 2.1, alpha = 0.9) +
    eig_scale_color(tokens = tokens, name = "Cylinders") +
    ggplot2::labs(
      title = "Fuel Economy by Vehicle Weight",
      subtitle = "EIG token-driven ggplot example",
      caption = "Synthetic example data"
    ) +
    eig_theme_ggplot(tokens = tokens)

  ggplot2::ggsave(
    filename = file.path(out_dir, "ggplot_scatter.png"),
    plot = p,
    width = tokens$EIG_FIGURE_DEFAULT_WIDTH_IN,
    height = tokens$EIG_FIGURE_DEFAULT_HEIGHT_IN,
    dpi = 220
  )
}

if (requireNamespace("plotly", quietly = TRUE) && requireNamespace("htmlwidgets", quietly = TRUE)) {
  p_int <- plotly::plot_ly(
    data = df,
    x = ~weight,
    y = ~mpg,
    color = ~cyl,
    colors = unname(tokens$EIG_DISCRETE),
    type = "scatter",
    mode = "markers"
  )
  layout_cfg <- eig_plotly_layout(tokens = tokens)
  p_int <- plotly::layout(
    p_int,
    title = "Fuel Economy by Vehicle Weight (Interactive)",
    xaxis = modifyList(layout_cfg$xaxis, list(title = "Weight")),
    yaxis = modifyList(layout_cfg$yaxis, list(title = "MPG")),
    font = layout_cfg$font,
    paper_bgcolor = layout_cfg$paper_bgcolor,
    plot_bgcolor = layout_cfg$plot_bgcolor,
    colorway = layout_cfg$colorway
  )
  htmlwidgets::saveWidget(
    p_int,
    file = file.path(out_dir, "plotly_scatter.html"),
    selfcontained = FALSE
  )
}

if (requireNamespace("gt", quietly = TRUE)) {
  summary_tbl <- aggregate(cbind(mpg, weight) ~ cyl, data = df, FUN = mean)
  summary_tbl$mpg <- round(summary_tbl$mpg, 2)
  summary_tbl$weight <- round(summary_tbl$weight, 2)
  gt_tbl <- gt::gt(summary_tbl)
  gt_tbl <- eig_style_gt(gt_tbl, tokens = tokens)
  gt::gtsave(gt_tbl, filename = file.path(out_dir, "gt_table.html"))
}

cat("Wrote R example outputs to:", out_dir, "\n")

File: Infrastructure/style/examples/python/generate_examples.py
#!/usr/bin/env python3
"""Generate EIG themed example outputs for Python."""

from __future__ import annotations

from pathlib import Path
import sys


REPO_ROOT = Path(__file__).resolve().parents[2]
sys.path.insert(0, str(REPO_ROOT))

try:
    import matplotlib.pyplot as plt
    import numpy as np
    import pandas as pd
    import seaborn as sns
    import plotly.express as px
except ImportError as exc:
    print(f"ERROR: Missing dependency: {exc}", file=sys.stderr)
    print("Install with: pip install matplotlib seaborn plotly pandas numpy", file=sys.stderr)
    raise SystemExit(2)

from themes.python.eig_theme import (
    assert_eig_fonts,
    eig_palette,
    eig_plotly_template,
    set_eig_theme,
    style_table,
)


def make_data(n: int = 140) -> pd.DataFrame:
    rng = np.random.default_rng(42)
    weight = rng.normal(3.1, 0.65, n).clip(1.5, 5.7)
    cyl = rng.choice([4, 6, 8], n, p=[0.4, 0.35, 0.25])
    origin = rng.choice(["Domestic", "Foreign"], n, p=[0.58, 0.42])
    mpg = 41.5 - 5.0 * weight - 0.75 * (cyl - 4) + rng.normal(0, 2.1, n)
    return pd.DataFrame(
        {
            "weight": weight.round(3),
            "mpg": mpg.round(2),
            "cyl": cyl,
            "origin": origin,
        }
    )


def main() -> int:
    out_dir = REPO_ROOT / "examples" / "outputs" / "python"
    out_dir.mkdir(parents=True, exist_ok=True)

    assert_eig_fonts(allow_fallback=True)
    set_eig_theme()
    df = make_data()

    fig, ax = plt.subplots()
    sns.scatterplot(
        data=df,
        x="weight",
        y="mpg",
        hue="cyl",
        palette=eig_palette("discrete"),
        ax=ax,
        alpha=0.9,
    )
    ax.set_title("Fuel Economy by Vehicle Weight", loc="left")
    ax.set_xlabel("Weight")
    ax.set_ylabel("MPG")
    ax.grid(axis="x", visible=False)
    fig.tight_layout()
    fig.savefig(out_dir / "matplotlib_scatter.png", dpi=220)
    plt.close(fig)

    summary = (
        df.groupby("origin", as_index=False)
        .agg(mean_mpg=("mpg", "mean"), mean_weight=("weight", "mean"))
        .round(2)
    )
    fig2, ax2 = plt.subplots()
    sns.barplot(
        data=summary,
        x="origin",
        y="mean_mpg",
        palette=eig_palette("discrete")[:2],
        ax=ax2,
    )
    ax2.set_title("Average MPG by Origin", loc="left")
    ax2.set_xlabel("")
    ax2.set_ylabel("Mean MPG")
    fig2.tight_layout()
    fig2.savefig(out_dir / "seaborn_bar.png", dpi=220)
    plt.close(fig2)

    fig_plotly = px.scatter(
        df,
        x="weight",
        y="mpg",
        color=df["cyl"].astype(str),
        title="Fuel Economy by Vehicle Weight (Interactive)",
        color_discrete_sequence=eig_palette("discrete"),
    )
    fig_plotly.update_layout(template=eig_plotly_template())
    fig_plotly.write_html(out_dir / "plotly_scatter.html", include_plotlyjs="cdn")

    table_df = (
        df.groupby("cyl", as_index=False)
        .agg(avg_mpg=("mpg", "mean"), avg_weight=("weight", "mean"))
        .round(2)
    )
    styled = style_table(table_df)
    (out_dir / "styled_table.html").write_text(styled.to_html(), encoding="utf-8")

    print(f"Wrote outputs to: {out_dir}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
